src/bin-fix-mismatches/fix-mismatches.ts

```diff
@@ -44,3 +55,18 @@ export function fixMismatches(ctx: Context): Context {
 
   return ctx;
 }
+
+function updateObjectNestedKeyFromPath(
+  obj: any,
+  nestedPropertyPath: string,
+  value: string | undefined,
+) {
+  const properties = nestedPropertyPath.split('.');
+  const lastKeyIndex = properties.length - 1;
+  for (let i = 0; i < lastKeyIndex; ++i) {
+    const key = properties[i];
+    if (!(key in obj)) obj[key] = {};
+    obj = obj[key];
+  }
+  obj[properties[lastKeyIndex]] = value;
+}
```

Comment:
This function should probably not be here but in a utils file somewhere, but I didn't know where to put it, so I let it here for now, waiting for instructions.

Reply:
Nice! yeah something like this will definitely be needed. 

We have [`props('path.t.a.value', isNumber)(someObject)`](https://github.com/JamieMason/syncpack/blob/e9fd5e4709634ffc83b0d876c0be8be2c9248308/src/lib/get-context/get-package-json-files/get-patterns/props.ts) ([tests here](https://github.com/JamieMason/syncpack/blob/e9fd5e4709634ffc83b0d876c0be8be2c9248308/src/lib/get-context/get-package-json-files/get-patterns/props.spec.ts)) for getting but nothing for setting. If you'd like to make this that would be great. 

I'm using https://mobily.github.io/ts-belt/api/result a lot which I appreciate is a style of coding that may not be to everyone's taste, so if you don't want to that's fine too.

Thanks a lot for this PR and your work so far, it's unlucky timing that it came right when I was doing a huge refactor. Things will be settled again now.